app/facades/nav_facade.rb

```diff
@@ -0,0 +1,26 @@
+class NavFacade < NavService
+
+  def self.get_directions(start_address, end_address)
+    route = route_directions(start_address, end_address)[:route][:legs][0][:maneuvers]
+
+    all_directions = [] 
+    turn_distances = []
+
+    route.each do |step|
+      all_directions << step[:narrative]
+      turn_distances << step[:distance]
+    end
+    all_directions.zip(turn_distances)
+  end
+
+  def self.trip_details(start_address, end_address)
+    details = route_directions(start_address, end_address)[:route].slice(:realTime, :fuelUsed, :hasTollRoad, :distance)
```

Comment:
In the future we may want to refactor this so we don't need to hit the API twice, since we already hit it once in `self.get_directions`, possibly can save that in a variable of some sort.

Reply:
Good Point